shared/modules/metametrics.js

```diff
@@ -167,6 +170,22 @@ export function getTrackMetaMetricsEvent(metamaskVersion, getDynamicState) {
     if (!event || !category) {
       throw new Error('Must specify event and category.')
     }
+    // Uses recursion to track a duplicate event with sensitive properties included,
+    // but metaMetricsId excluded
+    if (sensitiveProperties) {
```

Comment:
Should we add something like this to the condition, to prevent an unnecessary second event?

```suggestion
    if (!excludeMetaMetricsId && sensitiveProperties) {
```

Or throw an error if `excludeMetaMetricsId` is ever truthy when `sensitiveProperties` were passed in? :thinking:

Reply:
It would have prevented me from submitting a buggy PR, for sure!